### PR TITLE
Fix completion/linting/type checking with VSCode/pyright

### PR DESCRIPTION
**Pull request by ashb**
Pyright (the type engine powering VSCode's python extension) doesn't treat
`airflow` as a namespace package because of the `airflow/__init__.py` and it
doesn't want to/can't support detecting the `__path__ = ...` method of making
it an explicit namespace package, so we are left with no option but to create
Yet Another Stub File.

Tested by pytting `reveal_type(FAB_VERSION); reveal_type(TaskSDKDag)` inside
`_upgrade_outdated_dag_access_control` in `airflow/model/dag.py` -- before
this change it was reporting both as Unknown.

### CODE REVIEW

**Comment by ashb** [comment]
cc @JDarDagran

**Review by JDarDagran — APPROVED** [review]
As for VSCode user with Pylance enabled - it fixed the issue. Thank you a lot @ashb 🚀

**Comment by ashb** [comment]
Okay sadly we can't do this. By creating this empty file pycharm thinks `airflow` is an empty module

File: Dockerfile.ci [hunk 1]
```diff
@@ -660,7 +660,7 @@ function install_airflow() {
 
         # Similarly we need _a_ file for task_sdk too
         mkdir -p ./task_sdk/src/airflow/sdk/
-        touch ./task_sdk/src/airflow/__init__.py
+        touch ./task_sdk/src/airflow/sdk/__init__.py
```
Comment (ashb) [inline Dockerfile.ci:663]: This path was okay before, but since I added `exclude = ["src/airflow/__init__.py"]` in this PR this path now needs to change.

**Review by amoghrajesh — APPROVED** [review]
Pycharm test works as expected as discussed offline. LGTM

**Comment by ashb** [comment]
Static check failure unrelated and will be fixed by #43923

**Review by kaxil — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
Cool. Getting it right with PyCharm, VSCode namespace packages and static checks and CLI seems like an interesting game :)

**Review by pierrejeambrun — APPROVED** [review]

**Pull request merged**